var/spack/repos/builtin/packages/conduit/package.py

```diff
@@ -240,6 +241,56 @@ def check_install(self):
             example = Executable('./conduit_example')
             example()
 
+    def test(self):
+        """
+        Checks the spack install of conduit using conduit's
+        using-with-cmake and using-with-make examples
+        (supports: spack test run)
+        """
+        # Note: This looks similar to the `self.check_install` logic
+        # but there are subtle differences:
+        # (paths differ , and we use `self.run_test`)
+        print("Checking Conduit installation...")
+        spec = self.spec
+        install_prefix = spec.prefix
+        example_src_dir = join_path(install_prefix,
+                                    "examples",
+                                    "conduit",
+                                    "using-with-cmake")
+        print("Checking using-with-cmake example...")
+        with working_dir(join_path(self.install_test_root,
+                                   "check-post-conduit-using-with-cmake-example"),
+                         create=True):
+            opts = ["-DCONDUIT_DIR={0}".format(install_prefix), example_src_dir]
+            opts += self.std_cmake_args
+            #######
+            # TODO
+            #######
+            # how do we properly locate 'cmake' and 'make' in the
+            # `spack test run` environment ?
+            self.run_test("cmake", opts)
+            self.run_test("make", purpose="build example")
+            self.run_test("conduit_example", purpose="run example")
+        print("Checking using-with-make example...")
+        example_src_dir = join_path(install_prefix,
+                                    "examples",
+                                    "conduit",
+                                    "using-with-make")
+        example_files = glob.glob(join_path(example_src_dir, "*"))
+        with working_dir(join_path(self.install_test_root,
+                                   "check-post-conduit-using-with-make-example"),
+                         create=True):
+            for example_file in example_files:
+                shutil.copy(example_file, ".")
+            opts = ["CONDUIT_DIR={0}".format(install_prefix)]
+            #######
+            # TODO
+            #######
+            # how do we properly locate 'make' in the
+            # `spack test run` environment ?
+            self.run_test('make', opts, purpose="build example")
+            self.run_test('./conduit_example', purpose="build example")
+
```

Comment:
It would be prudent to remove or clean up the working directories -- since they are under the install prefix -- before (or after) the tests.  

Alternatively you could copy the directories/files to a subdirectory of `self.test_suite.stage` before building and running the tests.  Just keep in mind a test suite can run tests against multiple specs and or versions of the same spec.  (This will be automatically done for cached test sources once #23713 is merged so you could take a look at the changes to `package.py` there for how `llnl.util.filesystem`'s `install_tree` is being used for this purpose.)